server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/VariableWidthHistogramAggregator.java

```diff
@@ -293,17 +283,33 @@ public CollectionPhase collectValue(LeafBucketCollector sub, int doc, double val
                 collectBucket(sub, doc, numClusters - 1);
 
                 if(val > clusterCentroids.get(bucketOrd)){
-                    // Insert just ahead of bucketOrd so that the array remains sorted
+                    /*
+                     * If the new value is bigger than the nearest bucket then insert
+                     * just ahead of bucketOrd so that the array remains sorted.
+                     */
                     bucketOrd += 1;
                 }
                 moveLastCluster(bucketOrd);
+                updateAvgBucketDistanceIfModified(bucketOrd);
             } else {
                 addToCluster(bucketOrd, val);
                 collectExistingBucket(sub, doc, bucketOrd);
+                updateAvgBucketDistanceIfModified(bucketOrd);
             }
             return this;
         }
 
+        private void updateAvgBucketDistanceIfModified(int modifiedBucketOrd) {
+            if (modifiedBucketOrd == 0 || modifiedBucketOrd == numClusters - 1) {
```

Comment:
`numClusters` is the denominator of the `avgBucketDistance` equation. So I believe the distance should be updated whenever the number of clusters is changed, regardless of the position of this cluster.

In that case, maybe this whole method is necessary and `updateAvgBucketDistance()` should just be called every time?

This makes sense to me intuitively. If a bucket is added within the existing range of buckets, this should decrease the average bucket distance, since there are more buckets in the same range.

Reply:
++